imagedephi/redact/redact.py

```diff
@@ -35,8 +37,9 @@ def iter_image_files(directory: Path) -> Generator[Path, None, None]:
     May raise a PermissionError if the directory is not readable.
     """
     for child in directory.iterdir():
-        if child.suffix in FILE_EXTENSION_MAP:
-            yield child
+        if child.is_file() and os.access(child, os.R_OK):
+            if filetype.is_image(child):
+                yield child
```

Comment:
@brianhelba Is this a good way to handle the permissions errors that `filetype` is getting with mac and windows or should this be handled elsewhere?

Reply:
@brianhelba we had discussed this briefly but you had also mentioned possible better approaches. Let me know any suggestions you have.